src/main/webapp/src/app/recommendation-form/recommendation-form.component.ts

```diff
@@ -8,18 +9,25 @@ import { FormBuilder} from '@angular/forms';
 })
 export class RecommendationFormComponent{
 
+  @Output() formSubmit: EventEmitter<void> = new EventEmitter()
+
   // Build preference form
   preferenceForm = this.formBuilder.group({
-    price: "",
-    rating:"",
-    type: ""
+    price: ["", Validators.required],
+    rating: ["", Validators.required],
+    type: ["", Validators.required]
```

Comment:
I'm not sure what validators.required mean could you please put some comments or reply here

Reply:
That might help: https://angular.io/guide/reactive-forms#validating-form-input

Reply:
I added the Validators.required to ensure all inputs for the form are set :)